### PR TITLE
fix: bug in dashboard page width when link is too long

### PR DESCRIPTION
**Pull request by essofyany**
I there
here is a preview of the final result
Capture d’écran du 2023-12-19 22-15-28

### CODE REVIEW

**Comment by origranot** [comment]
This looks great!

What I had in my mind was to limit the visual url, subtract and show only the first 20/30~ characters of a link, we could add dots (...) to the suffix of the subtracted link.
In addition we could create a tooltip when hovering the subtracted link.

Your solution is great, however when you have multiple rows the height of each row is not fixed.
What do you think?

**Comment by essofyany** [comment]
Hello @origranot,

Great news! The adjustment ensures uniform row height, enhancing the table's overall cleanliness.

I attempted to integrate a tooltip, and the outcome is visible in the screenshot below:
Screenshot

However, I've noticed that for extremely long URLs, the tooltip expands proportionally, making it as extensive as the URL itself. This might impact the visual appeal. Any suggestions on how to manage this scenario would be appreciated.

**Comment by origranot** [comment]
> Hello @origranot,
> 
> Great news! The adjustment ensures uniform row height, enhancing the table's overall cleanliness.
> 
> I attempted to integrate a tooltip, and the outcome is visible in the screenshot below: Screenshot
> 
> However, I've noticed that for extremely long URLs, the tooltip expands proportionally, making it as extensive as the URL itself. This might impact the visual appeal. Any suggestions on how to manage this scenario would be appreciated.

This looks amazing!
I think we should allow more characters on the link itself, maybe extend in more 20/30 characters?
About the tooltip, I think we should use the same strategy. (most of the links won't be that long)
Keep in mind to support both light and dark mode 💯

**Comment by essofyany** [comment]
Hi @origranot,

Here are the results of my changes. The dark theme is already handled by DaisyUI.

Capture d’écran du 2023-12-22 13-23-09
Capture d’écran du 2023-12-22 13-29-36

I limited the link text in table column to 30 characters and the tooltip text to 150.
should we add tooltip to **Shortened URL** or we keep the tooltip for the long URL only?

File: apps/frontend/src/routes/dashboard/index.tsx [hunk 1]
```diff
@@ -38,9 +38,17 @@ export default component$(() => {
       displayName: 'Destination URL',
       classNames: 'w-1/4',
       format: $(({ value }) => {
+        const link = (limit: number) => (value.length > limit ? value.slice(0, limit) + '...' : value);
```
Comment (origranot) [inline apps/frontend/src/routes/dashboard/index.tsx:41]: Consider change the function name to be more accurate.
How about `limitLink`?

**Review by origranot — CHANGES_REQUESTED** [review]
Hey @essofyany,
This looks great, consider change the function name 💯

**Review by origranot — APPROVED** [review]
Nice!

**Pull request merged**